src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentValidationServiceImpl.java

```diff
@@ -126,6 +138,78 @@ private boolean isExplanationValid(String explanation, CreateAccountingDocumentR
         return explanationIsValid;
     }
     
+    private boolean documentWasParsedWithoutErrors(AccountingXmlDocumentEntry accountingXmlDocument,
+            CreateAccountingDocumentReportItemDetail reportItemDetail) {
+        if (CollectionUtils.isNotEmpty(accountingXmlDocument.getValidationErrors())) {
+            for (ValidationEvent validationError : accountingXmlDocument.getValidationErrors()) {
+                reportItemDetail.appendErrorMessageToExistingErrorMessage(
+                        buildValidationErrorMessageForUser(validationError));
+            }
+            LOG.info("documentWasParsedWithoutErrors: Detected one or more parsing validation errors.");
+            return false;
+        } else {
+            return true;
+        }
+    }
+    
+    private String buildValidationErrorMessageForUser(ValidationEvent validationError) {
+        String subMessage = getErrorSubMessage(validationError);
+        int lineNumber = getErrorLineNumber(validationError);
+        if (lineNumber != -1) {
+            return formatMessageFromProperty(CuFPKeyConstants.ERROR_CREATE_ACCOUNTING_DOCUMENT_XML_ADAPTER_ERROR,
+                    lineNumber, subMessage);
+        } else {
+            return subMessage;
+        }
+    }
+    
+    private String getErrorSubMessage(ValidationEvent validationError) {
+        String eventMessage = getErrorEventMessage(validationError);
+        LOG.warn("getErrorSubMessage: Detected validation error when parsing XML document: "
+                + eventMessage);
+        Matcher matcher = EXCEPTION_MESSAGE_PATTERN.matcher(eventMessage);
+        if (!matcher.matches()) {
+            return eventMessage;
+        }
+        
+        String exceptionClassname = matcher.group(EXCEPTION_CLASSNAME_GROUP);
+        if (StringUtils.isBlank(exceptionClassname)) {
+            throw new IllegalStateException(
+                    "Pattern matched message but it does not contain exception classname; this should NEVER happen!");
```

Comment:
Since we don't want to display the exception class on the report, do we need to throw this exception? perhaps just log the anomaly,

Reply:
Good point, I'll update this to log the situation instead of triggering a failure.